### PR TITLE
Initial commit for OPS

### PR DESCRIPTION
**Pull request by ekorchmar**
German procedures vocabulary (OPS) with some manual mappings and mostly automated translations

### CODE REVIEW

**Review by hardhouse — CHANGES_REQUESTED** [review]
1. incorrect format for load_stage
2. all mappings files (*.csv) MUST be on gdrive or on another repo

File: OPS/load_stage.sql [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+	pVocabularyDate			=> TO_DATE ('20200101', 'yyyymmdd'),
+	pVocabularyVersion		=> 'OPS Version 2020',
+	pVocabularyDevSchema	=> 'DEV_OPS'
+);
+END $_$
+;
+--1. Input source tables ops_src_agg and ops_mod_src for all years
+--WbImport example
+/*
+WbImport -file=/home/ekorchmar/Downloads/ops_all/ops2010.csv
```
Comment (dimshitc) [inline OPS/load_stage.sql:34]: remove this

Comment (dimshitc) [reply OPS/load_stage.sql:34]: this should go to manual work. we can create another folder where you store all manual work you need

File: OPS/load_stage.sql [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+	)
+select
+	code as concept_code,
+	full_term as concept_name_de,
+	valid_start_date,
+	valid_end_date
+from code_full_term
+where superclass like '%...%' -- down to lowest parental level for full name
+;
+--4. Feed automated translations in concept_stage_translated and fill in concept_stage
```
Comment (dimshitc) [inline OPS/load_stage.sql:180]: should be in manual work file

File: OPS/load_stage.sql [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+	h.superclass = a.concept_code
+;
+delete from concept_stage 
+where concept_code in (
+select concept_code from concept_stage
+group by concept_code
+having count (1) > 1) and
+invalid_reason is null
+;
+--7. Automated scripts
```
Comment (dimshitc) [inline OPS/load_stage.sql:245]: should step 7 be after step 8?

File: OPS/load_stage.sql [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+* Authors: Eduard Korchmae, Timur Vakhitov, Dmitry Dymshyts, Christian Reich
+* Date: 2020
+**************************************************************************/
+
+--0. Latest update construction
+DO $_$
+BEGIN
+	PERFORM VOCABULARY_PACK.SetLatestUpdate(
+	pVocabularyName			=> 'OPS',
+	pVocabularyDate			=> TO_DATE ('20200101', 'yyyymmdd'),
```
Comment (dimshitc) [inline OPS/load_stage.sql:25]: Is there any way to automate the date and version?

Comment (ekorchmar) [reply OPS/load_stage.sql:25]: Once we integrate automated file retrieval

File: OPS/load_stage.sql (hunk 1 above)
Comment (ekorchmar) [reply OPS/load_stage.sql:34]: This is example of source data import, not manual work

File: OPS/load_stage.sql (hunk 2 above)
Comment (ekorchmar) [reply OPS/load_stage.sql:180]: It is. Will update comments

**Review by dimshitc — APPROVED** [review]

**Review by hardhouse — APPROVED** [review]
very dirty
ps the site moved to www.bfarm.de

File: OPS/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+1. Download all ClaML files from https://www.dimdi.de/dynamic/de/klassifikationen/downloads/?dir=ops/ **for all years**
+2. Use Python processing script (OPS_convert.py) to extract source files and fill in the source tables. Append resulting tables to ops_src_agg and modifiers_append with version year as last field.
+3. Run FULL FastRecreate: SELECT devv5.FastRecreateSchema(main_schema_name=>'devv5', include_concept_ancestor=> false, include_deprecated_rels=> true, include_synonyms=> true);
```
[the remark this answers is not in the document]
Comment (m-khitrun) [reply OPS/README.md]: fixed

File: OPS/README.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+4. Run load_stage.sql; supply automated translations as needed in concept_manual.
+5. Run check_stage_tables function (should retrieve NULL): SELECT * FROM qa_tests.check_stage_tables();
+6. Run generic_update: DO $$ BEGIN PERFORM devv5.GenericUpdate(); END $$;
+7. Run basic tables check (should retrieve NULL): SELECT * FROM qa_tests.get_checks();
+8. Perform manual work described in the readme.md file in the 'manual_work' folder.
+
+Repeat steps 6-11. 
+
+9. Clear cache: SELECT * FROM qa_tests.purge_cache();
```
[the remark this answers is not in the document]
Comment (m-khitrun) [reply OPS/README.md]: done

File: OPS/manual_work/readme.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
+STEP 7 of the Refresh
+7.1. Upload concept_manual table into the working schema (skip this step if implementing on the Pallas vocabulary server).
+    Extract the [respective csv file](https://drive.google.com/drive/u/0/folders/1P2dJ9PDMDuu03K-EqzAR8QgmLj72kEB0) into the concept_manual table. The file was generated using the query:
```
[the remark this answers is not in the document]
Comment (m-khitrun) [reply OPS/manual_work/readme.md:3]: Actual.

File: OPS/README.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
+1. Download all ClaML files from https://www.dimdi.de/dynamic/de/klassifikationen/downloads/?dir=ops/ **for all years**
+2. Use Python processing script (OPS_convert.py) to extract source files and fill in the source tables. Append resulting tables to ops_src_agg and modifiers_append with version year as last field.
+3. Run FULL FastRecreate: SELECT devv5.FastRecreateSchema(main_schema_name=>'devv5', include_concept_ancestor=> false, include_deprecated_rels=> true, include_synonyms=> true);
+4. Run load_stage.sql; supply automated translations as needed in concept_manual.
+5. Run check_stage_tables function (should retrieve NULL): SELECT * FROM qa_tests.check_stage_tables();
```
[the remark this answers is not in the document]
Comment (m-khitrun) [reply OPS/README.md]: fixed

**Review by Alexdavv — APPROVED** [review]

**Pull request merged**